### ISSUE TITLE
SSH キーのトラブル

### ISSUE THREAD

**Issue by terasakisatoshi** [AtelierArith/OkinawaCompPhysFoodSurvey2024.jl#41]
https://docs.github.com/ja/authentication/connecting-to-github-with-ssh/checking-for-existing-ssh-keys ここを読んでもろて

**Comment by saintbenjamin** [comment]
僕もpushする際に、ちゃんとGitHubにSSH keyを登録したはずなのにそれを認識できず、IDやパスワードを聞きまくる現象が起こってます。

GitHubよりはGitLabの方を主に使ってて、GitLabの方はCLIコマンドでちゃんとプッシュできるのにGitHubはなぜかダメですね。

GitHub Desktopでボタンを押して何とかプッシュできました。

ちなみにGitHub DesktopですとGitHubのレポジトリのみならず、GitLabのレポジトリも同じGit系列なのでこれで一緒に使えます。

**Comment by terasakisatoshi** [comment]
https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent にある説明にあるように `~/.ssh/config` を生成した鍵のパスにすとできるはずです．

```
# ~/.ssh/config の中身
Host github.com
  AddKeysToAgent yes
  UseKeychain yes
  IdentityFile ~/.ssh/id_ed25519 # <---- ここを適宜編集
```

**Comment by saintbenjamin** [comment]
> https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent#adding-your-ssh-key-to-the-ssh-agent にある説明にあるように `~/.ssh/config` を生成した鍵のパスにすとできるはずです．
> 
> ```
> # ~/.ssh/config の中身
> Host github.com
>   AddKeysToAgent yes
>   UseKeychain yes
>   IdentityFile ~/.ssh/id_ed25519 # <---- ここを適宜編集
> ```

はい、そちらもそのようにしております。GitLabの方はちゃんと認識できてるのにGitHubの方はなぜかダメです。ただいまはTwo-Factor Authenticationを疑ってます。GitLabからは、そういう設定にしていてもAccess Tokenを発行してくれてそちらを入力すれば認証は普通に通ってますが、GitHubの方はそのようなものは多分ないようでですね。GitHub Desktopというアプリは、ちゃんと認証済みな状態になっているので、僕のCLIとは違っていつでもコミットやプッシュができてる感じです。GitLabの方はCLI上でもGitHub Desktop上でも両方できますけど。
